### PR TITLE
Fix initialize smart loan

### PR DESCRIPTION
**Pull request by mbare0**

### CODE REVIEW

File: contracts/PangolinExchange.sol [hunk 1]
```diff
@@ -62,7 +64,7 @@ contract PangolinExchange is OwnableUpgradeable, IAssetsExchange, ReentrancyGuar
    * @dev _minAvaxAmountOut minimum amount of the AVAX token to be bought
    **/
   function sellAsset(bytes32 _token, uint256 _exactERC20AmountIn, uint256 _minAvaxAmountOut) external override nonReentrant returns (bool) {
-    if (_exactERC20AmountIn == 0) revert InvalidTokenSaleAmount();
+    require(_exactERC20AmountIn != 0, "Amount of tokens to sell has to be greater than 0");
```
Comment (jakub-wojciechowski) [inline contracts/PangolinExchange.sol:67]: I think > 0, will suit here better

File: contracts/Pool.sol [hunk 2]
```diff
@@ -57,7 +57,7 @@ contract Pool is OwnableUpgradeable, ReentrancyGuardUpgradeable, IERC20 {
    **/
   function setRatesCalculator(IRatesCalculator ratesCalculator_) external onlyOwner {
     // setting address(0) ratesCalculator_ freezes the pool
-    if (!(Address.isContract(address(ratesCalculator_)) || address(ratesCalculator_) == address(0))) revert MustBeContract();
+    require(AddressUpgradeable.isContract(address(ratesCalculator_)) || address(ratesCalculator_) == address(0), "Must be a contract");
```
Comment (jakub-wojciechowski) [inline contracts/Pool.sol:60]: Hmm, the conditions on the left and right differ. It seems that the new one is correct. 
Do we have tests for that?

File: contracts/SmartLoan.sol [hunk 3]
```diff
@@ -35,12 +35,18 @@ contract SmartLoan is OwnableUpgradeable, PriceAwareUpgradeable, ReentrancyGuard
...
-  function initialize(IAssetsExchange assetsExchange_, Pool pool_) external initializer {
+  function initialize(IAssetsExchange assetsExchange_, Pool pool_, uint256 _initialDebt) external payable initializer {
     exchange = assetsExchange_;
     pool = pool_;
     __Ownable_init();
     __PriceAware_init();
     __ReentrancyGuard_init();
+
+    if (_initialDebt > 0) {
+      require(msg.value != 0, "No collateral provided for a loan");
```
Comment (jakub-wojciechowski) [inline contracts/SmartLoan.sol:46]: > 0 ?

File: contracts/SmartLoan.sol [hunk 4]
```diff
@@ -245,16 +251,30 @@ contract SmartLoan is OwnableUpgradeable, PriceAwareUpgradeable, ReentrancyGuard
...
+
     bytes32[] memory assets = exchange.getAllAssets();
 
-    for (uint256 i = 0; i < assets.length; i++) {
-      total = total + getAssetValue(assets[i]);
+    uint256[] memory prices = getPricesFromMsg(assets);
+
+    uint256 avaxPrice = prices[0];
+
+    require(avaxPrice != 0, "Avax price returned from oracle is zero");
```
Comment (jakub-wojciechowski) [inline contracts/SmartLoan.sol:266]: "from THE oracle"

File: contracts/SmartLoansFactory.sol [hunk 5]
```diff
@@ -43,43 +43,41 @@ contract SmartLoansFactory is OwnableUpgradeable, IBorrowersRegistry {
...
     updateRegistry(smartLoan);
     smartLoan.transferOwnership(msg.sender);
 
+    emit SmartLoanCreated(address(smartLoan), msg.sender, 0, 0);
+
     return smartLoan;
   }
 
   function createAndFundLoan(uint256 _initialDebt) external payable oneLoanPerOwner returns (SmartLoan) {
     BeaconProxy beaconProxy = new BeaconProxy(
```
Comment (jakub-wojciechowski) [inline contracts/SmartLoansFactory.sol:60]: Could we fit it in a single line

File: contracts/VariableUtilisationRatesCalculator.sol [hunk 6]
```diff
@@ -78,12 +78,9 @@ contract VariableUtilisationRatesCalculator is IRatesCalculator, Ownable {
       // unsigned integer safety check)
       uint256 value = (poolUtilisation * SLOPE_2) / 1e18 + MAX_RATE;
 
-      if (value < SLOPE_2) revert SlopeCalculationOutOfRange();
+      require(value >= SLOPE_2, "Out of range value when calculating borrowing rate. Consider checking if SLOPE_2 is calculated correctly");
```
Comment (jakub-wojciechowski) [inline contracts/VariableUtilisationRatesCalculator.sol:81]: calculating THE borrowing rate

File: contracts/deprecated/FixedRatesCalculator.sol [hunk 7]
```diff
@@ -25,7 +25,7 @@ contract FixedRatesCalculator is IRatesCalculator, Ownable {
    * @dev _borrowingRate the value of updated borrowing rate
    **/
   function setRates(uint256 _depositRate, uint256 _borrowingRate) public onlyOwner {
-    if (_depositRate > _borrowingRate) revert BorrowingRateLTDepositRate();
+    require(_depositRate <= _borrowingRate, "Borrowing rate cannot be lower than the deposit rate");
```
Comment (jakub-wojciechowski) [inline contracts/deprecated/FixedRatesCalculator.sol:28]: "A borrowing rate..."

File: contracts/SmartLoan.sol (hunk 3 above)
Comment (kavsky02) [reply contracts/SmartLoan.sol:46]: >0

**Pull request merged**